moped-editor/src/views/staff/components/NonLoginUserActivationButtons.js

```diff
@@ -0,0 +1,146 @@
+import React from "react";
+import { useNavigate } from "react-router-dom";
+import { useMutation } from "@apollo/client";
+import { UPDATE_NON_MOPED_USER } from "src/queries/staff";
+import {
+  useUserApi,
+  roleLooksGood,
+  passwordLooksGood,
+  fieldParsers,
+} from "../helpers";
+import { useButtonStyles } from "./StaffFormButtons";
+
+import { Button, Typography, Box } from "@material-ui/core";
+import clsx from "clsx";
+
+/**
+ * Generates a button to convert a non-login user to a Moped user with login
+ * @param {function} setModalState - set the modal's details
+ * @param {function} handleCloseModal - callback that fires on modal close
+ * @param {string} userId - moped_user primary key user_id
+ * @param {Object} formValues - User values from StaffForm
+ * @returns {JSX.Element}
+ */
+const NonLoginUserActivationButtons = ({
+  setModalState,
+  handleCloseModal,
+  userId,
+  formValues,
+}) => {
+  const classes = useButtonStyles();
+  let navigate = useNavigate();
+  const { email, password, roles } = formValues;
+
+  const { requestApi } = useUserApi();
+  const [updateNonMopedUser] = useMutation(UPDATE_NON_MOPED_USER);
+
+  /**
+   * Use the user activation API route to convert to a Moped user in the Cognito user pool with a DynamoDB claims entry
+   */
+  const handleNonLoginUserActivation = () => {
+    const rolesParser = fieldParsers["roles"];
+    const rolesArray = rolesParser("moped-editor");
+
+    const data = {
+      email,
+      password,
+      roles: rolesArray,
+    };
+
+    // Navigate to user table on success
+    const callback = () => navigate("/moped/staff");
+
+    requestApi({
+      method: "put",
+      path: "/users/activate/",
+      payload: data,
+      callback,
+    });
+  };
+
+  const handleUpdateNonLoginUser = () => {
+    const { password, ...restOfFormValues } = formValues;
+
+    updateNonMopedUser({
+      variables: {
+        userId: userId,
+        changes: restOfFormValues,
+      },
+    }).then(() => navigate("/moped/staff"));
+  };
+
+  const handleActivateNonLoginUserConfirm = () => {
+    if (!passwordLooksGood(password)) {
+      setModalState({
+        open: true,
+        title: "Error",
+        message: (
+          <Typography>
+            The password is required when activating a user. It needs to be 8
```

Comment:
It needs to be **12** characters long :)

Reply:
Thanks for catching this 🙏